xls/dslx/bytecode/bytecode_interpreter_test.cc

```diff
@@ -1253,6 +1253,68 @@ fn main() -> MyU9 {
   EXPECT_THAT(value.GetBitValueUint64(), IsOkAndHolds(0x1ff));
 }
 
+TEST(BytecodeInterpreterTest, AssertEqFailProcs) {
+  constexpr std::string_view kProgram = R"(
+#[test_proc]
+proc BTester {
+    terminator: chan<bool> out;
+
+    init { (u32:0) }
+
+    config(terminator: chan<bool> out) {
+        (terminator,)
+    }
+
+    next(tok: token, state: u32) {
+        let count = state;
+        let _ = assert_eq(count, u32:0);
+        let tok = send_if(tok, terminator, count > u32:1, true);
+        (count + u32:1)
+    }
+})";
+
+  XLS_ASSERT_OK_AND_ASSIGN(auto temp_file,
+                           TempFile::CreateWithContent(kProgram, "_test.x"));
+  constexpr std::string_view kModuleName = "test";
+  ParseAndTestOptions options;
+  options.handle_status = false;
+  absl::StatusOr<TestResult> result = ParseAndTest(
+      kProgram, kModuleName, std::string{temp_file.path()}, options);
+  EXPECT_THAT(result.status(), StatusIs(absl::StatusCode::kInternal,
```

Comment:
shouldn't this succeed? (i.e: if we do fix the bug we shouldn't get an internal error anymore)

Reply:
When assert expression is faulse, interpreter returns ststus that is `kInternal`.
I've taken this check from assert_eq test for functions(https://github.com/google/xls/blob/d86ff6c19790ddb8f390898c7814e77e4c0c05eb/xls/dslx/bytecode/bytecode_interpreter_test.cc#L212)